aws/resource_aws_securityhub_product_subscription.go

```diff
@@ -0,0 +1,85 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/securityhub"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSecurityHubProductSubscription() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSecurityHubProductSubscriptionCreate,
+		Read:   resourceAwsSecurityHubProductSubscriptionRead,
+		Delete: resourceAwsSecurityHubProductSubscriptionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"product_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func resourceAwsSecurityHubProductSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).securityhubconn
+	log.Printf("[DEBUG] Enabling Security Hub product subscription for product %s", d.Get("product_arn"))
+
+	resp, err := conn.EnableImportFindingsForProduct(&securityhub.EnableImportFindingsForProductInput{
+		ProductArn: aws.String(d.Get("product_arn").(string)),
+	})
+
+	if err != nil {
+		return fmt.Errorf("Error enabling Security Hub product subscription for product %s: %s", d.Get("product_arn"), err)
+	}
+
+	d.SetId(*resp.ProductSubscriptionArn)
+
+	return resourceAwsSecurityHubProductSubscriptionRead(d, meta)
+}
+
+func resourceAwsSecurityHubProductSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).securityhubconn
+
+	log.Printf("[DEBUG] Reading Security Hub product subscriptions to find %s", d.Id())
+	resp, err := conn.ListEnabledProductsForImport(&securityhub.ListEnabledProductsForImportInput{})
```

Comment:
The `ListEnabledProductsForImport` API call has a paginated counterpart in the AWS Go SDK, [`ListEnabledProductsForImportPages`](https://docs.aws.amazon.com/sdk-for-go/api/service/securityhub/#SecurityHub.ListEnabledProductsForImportPages), which we should use to prevent search issues in large environments, e.g.

```go
input := &securityhub.ListEnabledProductsForImportInput{}
productSubscriptionARN := ""

err := conn.ListEnabledProductsForImportPages(input, func(page *securityhub.ListEnabledProductsForImportOutput, lastPage bool) bool {
  for _, productSubscription := range resp.ProductSubscriptions {
    if aws.StringValue(productSubscription) == d.Id() {
      productSubscriptionARN = aws.StringValue(productSubscription)
      return false
    }
  }
  return !lastPage
})

if err != nil {
  return fmt.Errorf("Error reading Security Hub product subscriptions to find %s: %s", d.Id(), err)
}
```